replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/ReplicatorImpl.java

```diff
@@ -245,6 +288,16 @@ public void submitSyncRequest(final SyncRequest syncRequest) throws InterruptedE
     }
   }
 
+  @Override
+  public void cancelSyncRequest(SyncRequest syncRequest) {
+    pendingSyncRequests.remove(syncRequest);
```

Comment:
❓ Will this work for SyncRequest implementations that do not override equals and hashcode?

Reply:
I think this is fine for the time being. If we get to the point where we might have multiple implementations then we can revisit this.